tokio-util/src/time/delay_queue.rs

```diff
@@ -548,6 +548,8 @@ impl<T> DelayQueue<T> {
         let when = self.normalize_deadline(when);
 
         self.slab[key.index].when = when;
+        self.slab[key.index].expired = false;
```

Comment:
So, just to confirm, the reason this is correct is that the call to `insert_idx` will set it back to `true` if it is expired?

Reply:
Yes, `remove_key` combined with these assignments should leave the slab entry in the same state as a was newly created `Data` value in `insert_at`. Then `insert_idx` is responsible for placing the entry in the wheel or expired queue and setting the flag.